src/test/java/org/jenkinsci/plugins/gitclient/GitAPITestCase.java

```diff
@@ -405,17 +400,49 @@ private ObjectId getMirrorHead() throws IOException, InterruptedException
     protected String localMirror() throws IOException, InterruptedException {
         File base = new File(".").getAbsoluteFile();
         for (File f=base; f!=null; f=f.getParentFile()) {
-            if (new File(f,"target").exists()) {
-                File clone = new File(f, "target/clone.git");
-                if (!clone.exists()) {  // TODO: perhaps some kind of quick timestamp-based up-to-date check?
-                    w.launchCommand("git", "clone", "--mirror", "https://github.com/jenkinsci/git-client-plugin.git", clone.getAbsolutePath());
+            File targetDir = new File(f, "target");
+            if (targetDir.exists()) {
+                String cloneDirName = "clone.git";
+                File clone = new File(targetDir, cloneDirName);
+                if (!clone.exists()) {
+                    Path tempClonePath = Files.createTempDirectory(targetDir.toPath(), "clone-");
+                    w.launchCommand("git", "clone", "--reference", f.getCanonicalPath(), "--mirror", "https://github.com/jenkinsci/git-client-plugin", tempClonePath.toFile().getAbsolutePath());
```

Comment:
We can use the base directory as a reference repository because the base directory includes a `.git` directory which is a copy of the git client plugin repository.  

If the Jenkins job was configured with a narrow refspec or a shallow clone, it may be an imperfect copy, but it is a copy.  The copy, no matter how imperfect, can provide some objects for the reference repository.

Reply:
Oh yes, the base directory is the git client plugin directory, it would have the .git folder containing identical objects. 
Makes sense, thanks.